### PR TITLE
Add support for Self Signed certificates for email

### PR DESCRIPTION
**Pull request by Iandenh**
Add option to accept self signed certificates when connecting to smtp with CakeSocket.

Refs #6511

### CODE REVIEW

File: lib/Cake/Network/CakeSocket.php [hunk 1]
```diff
@@ -406,5 +406,19 @@ public function enableCrypto($type, $clientOrServer = 'client', $enable = true)
...
+/**
+ * Accept Self-signed certificate on current stream socket
+ *
+ * @return bool True on success
+ * @see stream_context_set_option
+ */
+	public function enableSelfSigned() {
+		$options['ssl'] = array(
+			'allow_self_signed' => true,
+			'verify_peer' => false,
```
Comment (markstory) [inline lib/Cake/Network/CakeSocket.php]: Why does allowing self signed need to turn off peer verification?

File: lib/Cake/Network/CakeSocket.php [hunk 2]
```diff
@@ -406,5 +406,14 @@ public function enableCrypto($type, $clientOrServer = 'client', $enable = true)
 		throw new SocketException($errorMessage);
 	}
 
+/**
+ * Accept Self-signed certificate on current stream socket
+ *
+ * @return bool True on success
+ * @see stream_context_set_option
```
Comment (ravage84) [inline lib/Cake/Network/CakeSocket.php]: ``@link http://php.net/manual/en/context.ssl.php About the 'allow_self_signed' option.`` might be better or good in addition.

File: lib/Cake/Network/CakeSocket.php [hunk 3]
```diff
@@ -406,5 +406,14 @@ public function enableCrypto($type, $clientOrServer = 'client', $enable = true)
...
+ */
+	public function enableSelfSigned() {
```
Comment (ravage84) [inline lib/Cake/Network/CakeSocket.php]: Can we have a test for this?

**Comment by markstory** [comment]
Looks good. I'll merge this in when I have a chance to write the 3.x equivalent.

**Comment by markstory** [comment]
Just looking at other pull requests I saw #7065 which aims to do a similar thing. Would you be against having options like.

``php
public $smtp = [
  'ssl_allow_self_signed' => true,
];
```
This would allow us to more easily support other ssl context options like verify_peer. The `HttpSocket` class has a [_configContext](https://github.com/cakephp/cakephp/blob/2.7/lib/Cake/Network/Http/HttpSocket.php#L710-L742) that could be moved down to CakeSocket and possibly solve this issue as well.

File: lib/Cake/Network/CakeSocket.php [hunk 4]
```diff
@@ -406,5 +406,15 @@ public function enableCrypto($type, $clientOrServer = 'client', $enable = true)
...
 	}
 
+/**
+ * Accept Self-signed certificate on current stream socket
+ *
+ * @return bool True on success
+ * @link http://php.net/manual/en/context.ssl.php About the 'allow_self_signed' option.
+ * @see stream_context_set_option
+ */
+	public function enableSelfSigned() {
```
Comment (ADmad) [inline lib/Cake/Network/CakeSocket.php]: "allowSelfSigned" would be a more appropriate name.

**Comment by markstory** [comment]
Thanks :+1: I did a few additional changes and have merged this in 94aeee438b8de135ace72742e4fbd873ba12af9c

**Pull request merged**